locked-nft/contracts/NFTLocker.cdc

```diff
@@ -147,7 +147,10 @@ pub contract NFTLocker {
                 self.lockedNFTs[nftType] <-! {}
             }
 
-            let oldToken <- self.lockedNFTs.insert(key: nftType, <-{id: <- token})
+            var lockedNftsByType <- self.lockedNFTs.remove(key: nftType)!
+            let oldToken <- lockedNftsByType.insert(key: id, <- token)
+            let oldDictionary <- self.lockedNFTs.insert(key: nftType, <- lockedNftsByType)
```

Comment:
@joshuahannan do you know if there is a better way to mutate `self.lockedNFTs`? I'm afraid this will not longer be performant as it grows. `self.lockedNFTs` is of type `@{Type: {UInt64: NonFungibleToken.NFT}}`

Reply:
the alternative is using a combination of the type and id as the dictionary key and the nft as the value. But i'll like to avoid a structural change if possible.

Reply:
Can you borrow a reference to it?

Reply:
worked like a charm